### ISSUE TITLE
Update Hono (bun) impl. for 20% speed boost

### ISSUE THREAD

**Issue by redbar0n** [SaltyAom/bun-http-framework-benchmark#59]
> For example, Hono on Bun looks like it performs badly. But just go to src/bun/hono.ts and change the export to export default { ...app, host: '127.0.0.1' } (instead of export default app) for an instant 20% speed boost that puts it right up, near the top.

according to https://github.com/hattipjs/hattip/issues/114#issuecomment-1884846698

**Comment by redbar0n** [comment]
Why benchmarks should use 127.0.0.1 instead of localhost: [don't use localhost (intermediate) anthony explains (4 min)](https://www.youtube.com/watch?v=98SYTvNw1kw).

**Comment by SaltyAom** [comment]
I agree with the benchmark of 127.0.0.1 benchmark part, but disagree with the hattip part:
> For example, Hono on Bun looks like it performs badly. But just go to src/bun/hono.ts and change the export to export default { ...app, host: '127.0.0.1' } (instead of export default app) for an instant 20% speed boost that puts it right up, near the top.

Basically, in any server more complex than hello world app, you are likely going to implemented either reverse proxy, micro-service or load balancing for multiple endpoint, eg. staging:

> api.stuff.dev
> staging.api.stuff.dev
> internal.api.stuff.dev

Setting the host ahead of time rejects all of the above implementation.

Additionally, the 20% speed boost is not only limited to Hono, but Elysia, Stric, Vixeny and most framework are going to benefits the speed boost as well since the hostname is know ahead of time, we can just omit the hostname entirely.

**Comment by redbar0n** [comment]
@cyco130

**Comment by cyco130** [comment]
> Setting the host ahead of time rejects all of the above implementation.

That was my point actually. When I ran the tests a couple of months ago, some of the tested frameworks assumed a certain host name (vixeny's `hasName`, stricjs's `base` etc.) which gave them an unfair advantage. Setting the host gives the same advantage to Hono. You can (and should) also go about the other way and disable those optimizations in those frameworks.

**Comment by SaltyAom** [comment]
Didn't realize Stric and Vixen hardcode basename.
Alright, going to remove the basename and rerun the benchmark right now, thanks.

**Comment by redbar0n** [comment]
@SaltyAom Would be awesome if you could update the README with the new bench results. It currently says `Tested on 18 Mar 2024 19:30 (GMT+7)`

**Comment by mimiMonads** [comment]
> > Setting the host ahead of time rejects all of the above implementation.
> 
> That was my point actually. When I ran the tests a couple of months ago, some of the tested frameworks assumed a certain host name (vixeny's `hasName`, stricjs's `base` etc.) which gave them an unfair advantage. Setting the host gives the same advantage to Hono. You can (and should) also go about the other way and disable those optimizations in those frameworks.

It actually doesn't matter, it has the same performance but the whole point of vixeny is purity, having the name of the server just set up things faster and avoids some FP complaining about it (the way I have to seal and encapsulate stuff)

Also this is the default option in `npx create-vixeny `

**Comment by mimiMonads** [comment]
> > Setting the host ahead of time rejects all of the above implementation.
> 
> That was my point actually. When I ran the tests a couple of months ago, some of the tested frameworks assumed a certain host name (vixeny's `hasName`, stricjs's `base` etc.) which gave them an unfair advantage. Setting the host gives the same advantage to Hono. You can (and should) also go about the other way and disable those optimizations in those frameworks.

@redbar0n 

Hello, I was looking at issues and I stumble upon this conversation, the reason why I added `hasName` (`base.bind` in the beta version)  is for  `vixeny-create` where I use it as a default point of reference, I'm removing it and I will talk to Reve to see if he can do the same.

It should be fixed by the next benchmark sorry for the trouble xd.

**Comment by aquapi** [comment]
@redbar0n Stric was abandoned for a while.
I will remove Stric and add my new tools.

**Comment by aquapi** [comment]
https://github.com/SaltyAom/bun-http-framework-benchmark/pull/71

**Comment by aquapi** [comment]
#71
